### PR TITLE
cli install import

### PR DESCRIPTION
**Pull request by will-moore**
Improve CLI install docs for when you need OMERODIR and server zip for ```omero import```

### CODE REVIEW

**Comment by jburel** [comment]
I think we need to add a bit more since the server installation guide also shows you how to install Java and Ice.
It won't work if it is not installed

**Comment by will-moore** [comment]
Same for python3.
But can we not assume that these are installed?
This is not quite the same as the server install docs where we want all the details to install from a clean machine. This page assumes you have python3 and I think it's OK to assume you have Java too (same as the Insight docs do).
I guess I could just add 'requires Java' into the text, but not cover how to install Java on all the various platforms.

We don't need to install Ice since it's covered by the ``pip install omoro-py``.

**Comment by jburel** [comment]
What I mean is that we should keep a link to the main installation page.

**Comment by pwalczysko** [comment]
>This is not quite the same as the server install docs where we want all the details to install from a clean machine.

Yes, but as @jburel says -> I still would benefit a lot by being linked to the docs install page, saying "if you do not. have python 3, then here it is how to install it"

File: omero/users/cli/installation.rst [hunk 1]
```diff
@@ -15,5 +15,16 @@ Since OMERO 5.6, only Python 3 is supported::
     omero login
 
 The |CLI| installed in this way provides all functionalities except import.
-Import requires jars in the OMERO.server package.
-See :doc:`/sysadmins/unix/server-installation`.
+Import requires jars in the OMERO.server package to be available under
```
Comment (mtbc) [inline omero/users/cli/installation.rst]:
```suggestion
Import requires JARs in the OMERO.server package to be available under
```

File: omero/users/cli/installation.rst [hunk 2]
```diff
@@ -15,5 +15,16 @@ Since OMERO 5.6, only Python 3 is supported::
...
+the ``$OMERODIR`` directory.
```
Comment (mtbc) [inline omero/users/cli/installation.rst]:
```suggestion
the :envvar:`OMERODIR` directory.
```
and elsewhere

**Comment by pwalczysko** [comment]
```
(venv) [root@5ebc31bfda5b /]# pip install omero-py
Collecting omero-py
  Using cached https://files.pythonhosted.org/packages/78/0b/2888ea12547b17058486e5b8c1e07ac4e3ec44c6cb63009bfc8f592230c3/omero-py-5.5.1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-v8r40vzp/omero-py/setup.py", line 16, in <module>
        from StringIO import StringIO
    ModuleNotFoundError: No module named 'StringIO'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-v8r40vzp/omero-py/
You are using pip version 9.0.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(venv) [root@5ebc31bfda5b /]# 
```

This does not work as written. In fact, I am blocked now, ran out of ideas.

Screen Shot 2020-01-08 at 10 41 14

**Comment by pwalczysko** [comment]
It would be good to have an overview of what I have to install and what I do not have to install.
Assuming I have to install

   - java and its prerequisites
 
I will test whether I have to install Ice and Ice python binding.

This should be specified in the CLI install doc.

**Comment by pwalczysko** [comment]
If I install Ice and ice python binding, then the import work with the latest artifact from py3-ci cc @joshmoore

**Comment by pwalczysko** [comment]
Clarified the situation now.
Principally, all works.

But, we would like to specify in the CLI client doc that  you have to install

   - java
   - ice
   - python3

And point to the server install page as we do.

The question remains whether or not we want to mention the possible problem with ``pip install omero-py`` as specified on https://openmicroscopy.slack.com/archives/CLHEW13NX/p1578483662017800 - @jburel clarified that this problem is remedied by using the 

```
# Install the Ice Python binding
$VENV_SERVER/bin/pip install https://github.com/ome/zeroc-ice-py-centos7/releases/download/0.2.1/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl
```
command, which is included in the server install walkthroughs we are pointing to anyway. Note that this cmd is OS specific, and the problem applies for several OS-es.

I can try to install python3 on my mac, but this will take some time yet.

**Comment by jburel** [comment]
@pwalczysko you could add that to the troubleshooting page
since the usage of the wheel is OS dependent

**Comment by pwalczysko** [comment]
All works on Mac too, except the problem https://openmicroscopy.slack.com/archives/CLHEW13NX/p1578491651051200 during ``pip install omeor-py>=5.6.dev``

This problem can be circumvented by using a conda workflow 

```
conda create -n myenv python=3.7
conda activate myenv
conda install -c ome zeroc-ice36-python 
pip install "omero-py>=5.6.dev"
```
from there, just follow the doc further, trying ``omero login``, downloading the server, unzipping, and importing -> this all works fine.

File: omero/users/cli/installation.rst [hunk 3]
```diff
@@ -4,16 +4,48 @@ Installation
...
+The ``omero`` command is now available::
+
     omero login
 
-The |CLI| installed in this way provides all functionalities except import.
-Import requires jars in the OMERO.server package.
-See :doc:`/sysadmins/unix/server-installation`.
+The |CLI| installed in this way provides all functionalities except the ``import`` functionality.
+
+The ``import`` functionality requires to have a supported version of Java e.g. Java, 11 Ice |iceversion| installed
```
Comment (mtbc) [inline omero/users/cli/installation.rst]:
```suggestion
The ``import`` functionality requires to have a supported version of Java e.g. Java 11, Ice |iceversion| installed
```

File: omero/users/cli/installation.rst [hunk 4]
```diff
@@ -4,16 +4,48 @@ Installation
...
+and also requires JARs in the OMERO.server package to be available under
+the :envvar:`OMERODIR` directory.
+
+To install Java and Ice, go to :doc:`/sysadmins/unix/server-installation` and select the walk-through corresponding to your OS.
```
Comment (mtbc) [inline omero/users/cli/installation.rst]:
```suggestion
To install Java and Ice, go to :doc:`/sysadmins/unix/server-installation`
and select the walkthrough corresponding to your OS.
```

File: omero/users/cli/installation.rst [hunk 5]
```diff
@@ -4,16 +4,48 @@ Installation
...
+
+To make the JARs available, you need to:
+
+* download the OMERO.server zip from under artifacts at :downloads:`downloads page <>`
```
Comment (mtbc) [inline omero/users/cli/installation.rst]: Maybe use numbers instead of bullets because this is an ordered sequence.

**Comment by will-moore** [comment]
Additional changes look good.

File: omero/users/cli/installation.rst [hunk 6]
```diff
@@ -4,16 +4,49 @@ Installation
...
-We recommend installing in a Python virtual environment.
-Since OMERO 5.6, only Python 3 is supported::
+Since OMERO 5.6, only Python 3 is supported.
+We assume that you have already installed Python 3.6.
 
-    python3.6 -mvenv venv
-    . venv/bin/activate
+We recommend installing omero-py and the |CLI| plugins
+in a Python virtual environment.
+You can either create one using ``venv`` or ``conda`` (preferred).
```
Comment (pwalczysko) [inline omero/users/cli/installation.rst]: ``You can either create one using...`` -> ``You can create one using either....``

File: omero/developers/Python.rst [hunk 7]
```diff
@@ -14,6 +16,19 @@ can be installed via ``pip``. We recommend you use a virtual environment:
...
+Install from Conda_:
+
+.. parsed-literal::
+
+    conda create -n myenv python=3.6
+    conda activate myenv
+    conda install -c ome zeroc-ice36-python 
+    conda install -c ome omero-py 
+
+You can then start using the library:
```
Comment (joshmoore) [inline omero/developers/Python.rst]: Is an activation needed here?

File: omero/developers/Python.rst [hunk 8]
```diff
@@ -5,7 +5,9 @@ To access the OMERO.server Python API, you need to install the Python client
 libraries.
 
 From OMERO 5.6.0 release, the client library ``omero-py`` supports Python 3 and
-can be installed via ``pip``. We recommend you use a virtual environment:
+is available on PyPI_ and Conda_. We recommend you use a virtual environment to install the client library:
```
Comment (joshmoore) [inline omero/developers/Python.rst]: The virtual environment is only if not conda, correct?

File: omero/users/cli/installation.rst (hunk 6 above)
Comment (joshmoore) [inline omero/users/cli/installation.rst]: :+1: This terminology works better for me than the one above in Python.rst.

File: omero/users/cli/installation.rst [hunk 9]
```diff
@@ -4,16 +4,49 @@ Installation
...
+If you opt for Conda_, you will need
+to install it first, see `miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ for more details.
+
+To create a virtual env and install ``omero-py`` using venv::
+
+    python3.6 -m venv py3_venv
```
Comment (joshmoore) [inline omero/users/cli/installation.rst]: Perhaps s/py3_venv/myenv/ so there's no confusion why the name is different.

File: omero/users/cli/installation.rst [hunk 10]
```diff
@@ -4,16 +4,49 @@ Installation
...
+
+#. download the OMERO.server zip from under artifacts at :downloads:`downloads page <>`
+#. unzip the zip file 
+#. set ``$OMERODIR`` to the unzipped directory::
+
+    export OMERODIR=/path/to/OMERO.server-x.x.x-ice36-bxx
+
+The ``import`` functionality is now available::
+
+    omero import /path/to/image.tiff
```
Comment (joshmoore) [inline omero/users/cli/installation.rst:49]: Worth mentioning here that we'll be looking to improve this workflow?

**Review by joshmoore — COMMENTED** [review]
Few minor thoughts.

File: omero/users/cli/installation.rst [hunk 11]
```diff
@@ -4,16 +4,49 @@ Installation
...
-    # omero command is now available
+To create a virtual env and install ``omero-py`` using conda (preferred)::
+
+    conda create -n myenv python=3.6
+    conda activate myenv
+    conda install -c ome zeroc-ice36-python 
+    conda install -c ome omero-py 
+
+The ``omero`` command is now available::
+
```
Comment (pwalczysko) [inline omero/users/cli/installation.rst:28]: Tested this precise workflow on my mac -> pass

File: omero/users/cli/installation.rst [hunk 12]
```diff
@@ -4,16 +4,49 @@ Installation
...
     omero login
 
-The |CLI| installed in this way provides all functionalities except import.
-Import requires jars in the OMERO.server package.
-See :doc:`/sysadmins/unix/server-installation`.
+The |CLI| installed in this way provides all functionalities except the ``import`` functionality.
+
+The ``import`` functionality requires to have a supported version of Java e.g. Java 11, Ice |iceversion| installed
```
Comment (pwalczysko) [inline omero/users/cli/installation.rst]: Replace with ``...requires to have a supported version of Java e.g. Java 11 and Ice (iceversion) installed. Also, it requires JARs....`` (better wording, at the moment, the Ice is a bit drowned in the sentence)

File: omero/users/cli/installation.rst [hunk 13]
```diff
@@ -4,16 +4,49 @@ Installation
...
+and also requires JARs in the OMERO.server package to be available under
+the :envvar:`OMERODIR` directory.
+
+To install Java and Ice, go to :doc:`/sysadmins/unix/server-installation`
+and select the walkthrough corresponding to your OS.
+
+To make the JARs available, you need to:
+
+#. download the OMERO.server zip from under artifacts at :downloads:`downloads page <>`
```
Comment (pwalczysko) [inline omero/users/cli/installation.rst]: Would it be better to point to the https://www.openmicroscopy.org/omero/downloads/ ? It is a bit hard to navigate through the artifacts

Comment (will-moore) [reply omero/users/cli/installation.rst]: Done

File: omero/developers/Python.rst [hunk 14]
```diff
@@ -14,6 +16,19 @@ can be installed via ``pip``. We recommend you use a virtual environment:
...
+    conda install -c ome zeroc-ice36-python 
+    conda install -c ome omero-py 
+
+You can then start using the library:
+
+.. parsed-literal::
+
     $ python
     >>> from omero.gateway import BlitzGateway
     >>> conn = BlitzGateway('username', 'password', host='omero.server', port=4064)
```
Comment (pwalczysko) [inline omero/developers/Python.rst:34]: tested this snippet on mac after the conda install. Works as expected.

File: omero/users/cli/installation.rst (hunk 10 above)
Comment (pwalczysko) [reply omero/users/cli/installation.rst:49]: Wouldn't do that. The workflow is not that bad at all - if mention, then much higher, where the "download the server" instruction is posing.

File: omero/developers/Python.rst (hunk 8 above)
Comment (pwalczysko) [reply omero/developers/Python.rst]: I took it as if conda is also creating a venv, only conda-style-venv - is it a wrong take on the matters ? If yes, then reformulation is indeed needed.

**Comment by pwalczysko** [comment]
All is fine, I could not test in the Centos7 docker, as conda init is not working there - I guess I would have to have the conda lines in the dockerfile, but the dockerfile I am using is not in my hand.

Except the comments of @joshmoore , all looks good to me.

File: omero/developers/Python.rst [hunk 15]
```diff
@@ -14,6 +16,19 @@ can be installed via ``pip``. We recommend you use a virtual environment:
     $ pip install omero-py==\ |version_py|
     Successfully installed Pillow-6.2.1 future-0.18.2 omero-py-5.6.1 zeroc-ice-3.6.5
 
+Install from Conda_:
+
+.. parsed-literal::
+
+    conda create -n myenv python=3.6
```
Comment (manics) [inline omero/developers/Python.rst]: You can simplify all this to one line
```suggestion
    conda create -n myenv -c ome python=3.6 omero-py
```

File: omero/developers/Python.rst [hunk 16]
```diff
@@ -14,6 +16,19 @@ can be installed via ``pip``. We recommend you use a virtual environment:
...
+    conda activate myenv
+    conda install -c ome zeroc-ice36-python
```
Comment (manics) [inline omero/developers/Python.rst]:
```suggestion
```

File: omero/developers/Python.rst [hunk 17]
```diff
@@ -14,6 +16,19 @@ can be installed via ``pip``. We recommend you use a virtual environment:
...
     Successfully installed Pillow-6.2.1 future-0.18.2 omero-py-5.6.1 zeroc-ice-3.6.5
 
+Install from Conda_:
+
+.. parsed-literal::
+
+    conda create -n myenv python=3.6
+    conda activate myenv
+    conda install -c ome zeroc-ice36-python 
+    conda install -c ome omero-py
```
Comment (manics) [inline omero/developers/Python.rst]:
```suggestion
```

File: omero/developers/Python.rst (hunk 15 above)
Comment (manics) [reply omero/developers/Python.rst]: You can optionally drop the `python=3.6` and you'll automatically get a version that's compatible with omero-py

Comment (joshmoore) [reply omero/developers/Python.rst]: TIL :+1:

**Review by pwalczysko — APPROVED** [review]

**Pull request merged**